### DISCUSSION TITLE
Cannot access SOCA API using AWS Lambda Function using Python code.

### DISCUSSION THREAD

**Discussion by balu-hcl** [awslabs/scale-out-computing-on-aws#122] [Q&A] [answered]
We are trying connect soca using SOCA API. We used api key to establish the communication, but we receive "not authorized error".
https://wtgsiesd-soca.design-tech.a2z.com/api/doc/
We are trying to reach through AWS Lambda function.

But same connection can be executed within a SOCA Desktop Session terminal.

**Comment by mcrozes** [comment, marked as answer]
Hello @balu-hcl -

This environment is owned and managed by another Amazon internal team. I will follow-up with you directly on your Amazon email.

Generally speaking, if you are trying to access to the SOCA API endpoint via a Lambda you just created, you have to verify if your Lambda is deployed on the same VPC or using generic environment (this will depends whether you should reference the API endpoint via the private or public endpoint).

If you are getting Not Authorized, the most obvious reason would be that the user you are initiating the connection with is not authenticated correctly and/or you are trying to access an admin-only API with a non-admin user.  Here is an example of simple API verification: https://awslabs.github.io/scale-out-computing-on-aws/web-interface/control-hpc-job-with-http-web-rest-api/#step1-retrieve-your-api-key
